src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApiBehaviorOptionsSetup.cs

```diff
@@ -45,11 +29,8 @@ public void Configure(ApiBehaviorOptions options)
             ConfigureClientErrorMapping(options);
         }
 
-        public override void PostConfigure(string name, ApiBehaviorOptions options)
+        public void PostConfigure(string name, ApiBehaviorOptions options)
         {
-            // Let compatibility switches do their thing.
```

Comment:
In 3.0, `InvalidModelStateResponseFactory` would default to returning a `ProblemDetails` instance (i.e. `options.InvalidModelStateResponseFactory = ProblemDetailsFactory;`) and `SuppressMapClientErrors` would have no bearing on it. Any user configured factory would take precedence over the default

Reply:
@pranavkm I don't disagree. But, are you asking for a change here?

Reply:
Basically, set `options.InvalidModelStateResponseFactory = ProblemDetailsFactory;` in `Configure` and remove the `PostConfigure` since it's no longer conditional.

Reply:
The `InvalidModelStateResponseFactory` setting remains conditional on `SuppressMapClientErrors` and that's a regular option that may be set at any point. Better to leave reading it 'til `PostConfigure(...)`.